api/bolt/migrator/migrate_dbversion22.go

```diff
@@ -49,6 +49,16 @@ func (m *Migrator) updateEndpointsAndEndpointGroupsToDBVersion23() error {
 		if err != nil {
 			return err
 		}
+
+		relation := &portainer.EndpointRelation{
```

Comment:
Is it actually required to create the relation in for all existing endpoints and not only when needed (e.g. only when a endpoint actually have associated stacks)?

Reply:
if we don't create it, we will need to check the error every time we try to get the relation object. I think it's safer this way